coral-common/src/main/java/com/linkedin/coral/common/functions/OperandTypeInference.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2018-2021 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.functions;
+
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+
+// TODO: move this to calcite code
```

Comment:
do we really mean it ? if so, let's create a issue / jira there and link it here ?

Reply:
I think we are not planning to do it now, removed,